src/options/pages/blueprints/modals/ShareRecipeModal/PublishRecipeModal.tsx

```diff
@@ -0,0 +1,232 @@
+
+const PublishRecipeModal: React.FunctionComponent = () => {
+  const dispatch = useDispatch();
+  const { blueprintId } = useSelector(selectShowPublishContext);
+  const { scope: userScope, organizations: userOrganizations } =
+    useSelector(selectAuth);
+  const [updateRecipe] = useUpdateRecipeMutation();
+  const { data: editablePackages, isFetching: isFetchingEditablePackages } =
+    useGetEditablePackagesQuery();
+  const {
+    data: recipe,
+    isFetching: isFetchingRecipe,
+    refetch: refetchRecipes,
+  } = useRecipe(blueprintId);
+
+  const [isPublishing, setPublishing] = React.useState(false);
+  const [error, setError] = React.useState<string | null>(null);
+
+  const closeModal = () => {
+    dispatch(blueprintModalsSlice.actions.closeModal());
+  };
+
+  // If the was just converted to a blueprint, the API request is likely be in progress and recipe will be null
+  if (isFetchingRecipe) {
+    return <Loading />;
+  }
+
+  const publish = async () => {
+    setPublishing(true);
+    setError(null);
+
+    try {
+      const newRecipe = produce(recipe, (draft) => {
+        draft.sharing.public = true;
+      });
+
+      const packageId = editablePackages.find(
+        (x) => x.name === newRecipe.metadata.id
+      )?.id;
+
+      await updateRecipe({
+        packageId,
+        recipe: newRecipe,
+      }).unwrap();
+
+      notify.success("Shared brick");
+      closeModal();
+      refetchRecipes();
+    } catch (error) {
+      if (
+        isSingleObjectBadRequestError(error) &&
+        error.response.data.config?.length > 0
+      ) {
+        setError(error.response.data.config.join(" "));
+        return;
+      }
+
+      const message = getErrorMessage(error);
+      setError(message);
+
+      notify.error({
+        message,
+        error,
+      });
+    } finally {
+      setPublishing(false);
+    }
+  };
+
+  // Sorting returns new array, so it safe to mutate it
+  const sortedOrganizations = sortOrganizations(userOrganizations);
+  const [recipeScope] = getScopeAndId(recipe?.metadata.id);
+  let ownerLabel: ReactElement;
+  let hasEditPermissions = false;
+  if (recipeScope === userScope) {
+    ownerLabel = (
+      <span>
+        <FontAwesomeIcon icon={faUser} /> You
+      </span>
+    );
+    hasEditPermissions = true;
+  } else {
+    const ownerOrganizationIndex = sortedOrganizations.findIndex(
+      (x) => x.scope === recipeScope
+    );
+
+    if (ownerOrganizationIndex === -1) {
+      ownerLabel = (
+        <span>
+          <FontAwesomeIcon icon={faUsers} /> Unknown
+        </span>
+      );
+    } else {
+      const ownerOrganization = sortedOrganizations.splice(
+        ownerOrganizationIndex,
+        1
+      )[0];
+      ownerLabel = (
+        <span>
+          <FontAwesomeIcon icon={faUsers} /> {ownerOrganization.name}
+        </span>
+      );
+      hasEditPermissions = editorRoles.has(ownerOrganization?.role);
+    }
```

Comment:
It seems like this logic/code could be cleaned up a bit:

```suggestion
    const ownerOrganization = sortedOrganizations.find(
      (x) => x.scope === recipeScope
    );
    
    hasEditPermissions = ownerOrganization && editorRoles.has(ownerOrganization?.role);
    ownerLabel = (
      <span>
          <FontAwesomeIcon icon={faUsers} /> {ownerOrganization?.name ?? "Unknown"}
        </span>
    );
```

Reply:
I don't think so. The catch is that `splice` operation mutates the `sortedOrganizations`. I'll take a closer look though and refactor or add a comment.

Reply:
Ah that's true, although it seems weird to pull out that one item and then render it at the top of the list anyway. It's fine for now for this PR, we don't have to rework the entire thing right now.

L210:
```
<div className={styles.row}>
  {ownerLabel}
  <span className="text-muted">Owner</span>
</div>
{sortedOrganizations
  .filter((x) =>
    recipe.sharing.organizations.includes(x.id as UUID)
  )
  .map((organization) => (
    <div className={styles.row} key={organization.id}>
      <span>
        <FontAwesomeIcon icon={faUsers} /> {organization.name}
      </span>
    </div>
  ))}
```